ownCloud/SceneDelegate.swift

```diff
@@ -0,0 +1,108 @@
+ *
+ */
+
+import UIKit
+import ownCloudSDK
+
+@available(iOS 13.0, *)
+class SceneDelegate: UIResponder, UIWindowSceneDelegate {
+
+	var window: UIWindow?
+
+	// UIWindowScene delegate
+	func scene(_ scene: UIScene, willConnectTo session: UISceneSession, options connectionOptions: UIScene.ConnectionOptions) {
+		if let windowScene = scene as? UIWindowScene {
+			window = UIWindow(windowScene: windowScene)
+			let serverListTableViewController = ServerListTableViewController(style: UITableView.Style.plain)
+			serverListTableViewController.restorationIdentifier = "ServerListTableViewController"
+			let navigationController = ThemeNavigationController(rootViewController: serverListTableViewController)
+			window?.rootViewController = navigationController
+			window?.addSubview((navigationController.view)!)
+			window?.makeKeyAndVisible()
+		}
+
+		if let userActivity = connectionOptions.userActivities.first ?? session.stateRestorationActivity {
+			if !configure(window: window, with: userActivity) {
+			}
+		}
+	}
+
+	func stateRestorationActivity(for scene: UIScene) -> NSUserActivity? {
+		return scene.userActivity
+	}
+
+	func configure(window: UIWindow?, with activity: NSUserActivity) -> Bool {
+		if activity.title == ownCloudOpenAccountPath {
+			if let bookmarkUUIDString = activity.userInfo?[ownCloudOpenAccountAccountUuidKey] as? String, let bookmarkUUID = UUID(uuidString: bookmarkUUIDString), let bookmark = OCBookmarkManager.shared.bookmark(for: bookmarkUUID) {
+				if let navigationController = window?.rootViewController as? ThemeNavigationController, let serverListController = navigationController.topViewController as? ServerListTableViewController {
+					serverListController.connect(to: bookmark, animated: false) { (_, _) in
+					}
+					window?.windowScene?.userActivity = bookmark.openAccountUserActivity
+					return true
+				}
+			}
+		} else if activity.title == ownCloudOpenItemPath {
+			if let itemLocalID = activity.userInfo?[ownCloudOpenItemUuidKey] as? String, let bookmarkUUIDString = activity.userInfo?[ownCloudOpenAccountAccountUuidKey] as? String, let bookmarkUUID = UUID(uuidString: bookmarkUUIDString), let bookmark = OCBookmarkManager.shared.bookmark(for: bookmarkUUID) {
+				if let navigationController = window?.rootViewController as? ThemeNavigationController, let serverListController = navigationController.topViewController as? ServerListTableViewController {
+
+					serverListController.connect(to: bookmark, animated: false) { (completed, clientRootViewController) in
+						if completed, let clientViewController = clientRootViewController.filesNavigationController?.topViewController as? ClientQueryViewController, let core = clientRootViewController.core {
+							core.retrieveItemFromDatabase(forLocalID: itemLocalID, completionHandler: { (error, _, item) in
+								if error == nil, let item = item {
+									OnMainThread {
+										var parentItems = core.retrieveParentItems(for: item)
```

Comment:
@hosy may be add some comments of what is happening here? Eventually extract a function? Sth. like ClientRootViewController.clone()?

Reply:
@mneuwert moved the code into ClientRootViewController